documentation/blog/2024-08-20-google-summer-of-code.md

```diff
@@ -0,0 +1,189 @@
+```
+This command creates a new Nmap scan targeting `scanme.nmap.org`.
+
+**Output:**
+```bash
+🆕 Creating a new scan with name 'nmap' and parameters 'scanme.nmap.org'
+🚀 Successfully created a new Scan 'nmap'
+```
+
+#### **2. Observe Scan Command (`scbctl scan --follow`)**
+
+The `--follow` flag enhances the `scbctl scan` command by providing real-time feedback on the progress of a scan. Once a scan is initiated, users can observe its progress directly from their terminal. This feature interacts with the secureCodeBox operator by streaming logs from the Kubernetes Job and Pods associated with the scan, giving users visibility into the scan’s status and results as they happen.
+
+**Usage Example:**
+```bash
+scbctl scan nmap --follow -- scanme.nmap.org
+```
+This command initiates a scan and follows its progress in real-time.
+
+**Output:**
+```bash
+Found 1 job(s)
+Job: scan-nmap-jzmtq, Labels: map[securecodebox.io/job-type:scanner]
+scan-nmap-jzmtq📡 Streaming logs for job 'scan-nmap-jzmtq' and container 'nmap'
+Starting Nmap 7.95 ( https://nmap.org ) at 2024-08-23 11:59 UTC
+Nmap scan report for scanme.nmap.org (45.33.32.156)
+Host is up (0.33s latency).
+Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
+Not shown: 996 closed tcp ports (conn-refused)
+PORT      STATE    SERVICE
+22/tcp    open     ssh
+80/tcp    filtered http
+9929/tcp  open     nping-echo
+31337/tcp open     Elite
+
+Nmap done: 1 IP address (1 host up) scanned in 30.19 seconds
+```
+
+#### **3. Trigger Scan Command (`scbctl trigger`)**
+
+The `scbctl trigger` command allows users to manually trigger a `ScheduledScan` resource. Scheduled scans are designed to run at predefined intervals, but there are times when an immediate execution is required. This command interacts with the secureCodeBox operator by invoking the `ScheduledScan` resource and creating a new `Scan` based on the schedule’s configuration.
+
+**Usage Example:**
+```bash
+scbctl trigger nmap --namespace foobar
+```
+This command triggers the `nmap` scheduled scan immediately.
+
+**Output:**
+```bash
+triggered new Scan for ScheduledScan 'nmap'
+```
+
+#### **4. Cascade Visualization Command (`scbctl cascade`)**
+
+The `scbctl cascade` command provides a visualization of cascading scans—scans that are automatically triggered based on the results of a previous scan. This command interacts with the secureCodeBox operator by querying all `Scan` resources in a given namespace and identifying relationships based on the `ParentScanAnnotation`. It then generates a hierarchical tree that visually represents these cascading relationships.
+
+**Usage Example:**
+```bash
+scbctl cascade
+```
+This command visualizes the cascading relationships between scans in the current namespace.
+
+**Output:**
+```bash
+Scans
+├── initial-nmap-scan
+│   ├── follow-up-vulnerability-scan
+│   │   └── detailed-sql-injection-scan
+└── another-initial-scan
+    └── another-follow-up-scan
+```
```

Comment:
Hi, @Freedisch  ! Thank you for using [gtree](https://github.com/ddddddO/gtree) for scbctl cascade! I'm very happy. (I'm the author of gtree)

I found this PullRequest and took a quick look, but maybe the output below is correct?

```bash
Scans
├── initial-nmap-scan
│   └── follow-up-vulnerability-scan
│       └── detailed-sql-injection-scan
└── another-initial-scan
    └── another-follow-up-scan
```

Also, I read this blog. Excellent!
If you want to improve the cascade subcommand, maybe [**WalkProgrammably** func](https://github.com/ddddddO/gtree?tab=readme-ov-file#walkprogrammably-func) is more useful than OutputProgrammably func.

The following link, which collects some tree libraries (and commands) other than gtree, might be helpful!
https://github.com/stars/ddddddO/lists/tree

Sorry for appearing out of nowhere! Bye!

Reply:
Thanks for the feedback : )
I will definitely take a look at the `WalkProgrammably func`